www/js/init.js

```diff
@@ -33,17 +33,168 @@
+    if (params.referrerExtensionURL && ~window.location.href.indexOf(params.PWAServer)) {
+        var message = '?PWA_launch=success';
+        // DEV: To test failure of the PWA, you could pause on next line and set message to '?PWA_launch=fail'
+        // Note that, as a failsafe, the PWA_launch key is set to 'fail' (in the extension) before each PWA launch
+        // so we need to send a 'success' message each time the PWA is launched
+        var frame = document.createElement('iframe');
+        frame.id = 'kiwixComm';
+        frame.style.display = 'none';
+        document.body.appendChild(frame);
+        frame.src = params.referrerExtensionURL + '/www/index.html' + message;
+        // Now remove redundant frame. We cannot use onload, because it doesn't give time for the script to run.
+        setTimeout(function () {
+            var kiwixComm = document.getElementById('kiwixComm');
+            // The only browser which does not support .remove() is IE11, but it will never run this code
+            if (kiwixComm) kiwixComm.remove();
+        }, 3000);
+    }
+})();
+
+// Since contentInjectionMode can be overriden when returning from remote PWA to extension (for example), we have to prevent an infinite loop
+// with code that warns the user to turn off the App Cache bypass in jQuery mode. Note that to turn OFF the bypass, we have to set the VALUE to true
+params.appCache = params.contentInjectionMode === 'jquery' ? true : params.appCache;
+
+/**
+ * Set the State and UI settings associated with parameters defined above
+ */
+document.getElementById('hideActiveContentWarningCheck').checked = params.hideActiveContentWarning;
+document.getElementById('disableDragAndDropCheck').checked = params.disableDragAndDrop;
+document.getElementById('showUIAnimationsCheck').checked = params.showUIAnimations;
+document.getElementById('titleSearchRange').value = params.maxSearchResultsSize;
+document.getElementById('titleSearchRangeVal').textContent = params.maxSearchResultsSize;
+document.getElementById('appThemeSelect').value = params.appTheme;
+document.getElementById('useHomeKeyToFocusSearchBarCheck').checked = params.useHomeKeyToFocusSearchBar;
+document.getElementById('openExternalLinksInNewTabsCheck').checked = params.openExternalLinksInNewTabs;
+document.getElementById('bypassAppCacheCheck').checked = !params.appCache;
+document.getElementById('appVersion').textContent = 'Kiwix ' + params.appVersion;
+
+var keyPrefix = params.keyPrefix;
+
+function getSetting (name) {
+    var result;
+    if (params.storeType === 'cookie') {
+        var regexp = new RegExp('(?:^|;)\\s*' + name + '=([^;]+)(?:;|$)');
+        result = document.cookie.match(regexp);
+        result = result && result.length > 1 ? decodeURIComponent(result[1]) : null;
+    } else if (params.storeType === 'local_storage') {
+        // Use localStorage instead
+        result = localStorage.getItem(keyPrefix + name);
```

Comment:
If you correct the code as I have suggested, then you can get rid of the `keyPrefix` definition and just use `params.keyPrefix` here and below.